artifactory.py

```diff
@@ -1730,6 +1753,32 @@ def deploy_file(
                 explode_archive_atomic=explode_archive_atomic,
             )
 
+    def deploy_by_checksum(
+        self,
+        sha1=None,
+        sha256=None,
```

Comment:
```suggestion
        sha256_val="",
```

Reply:
Hi @beliaev-maksim 

Renaming looks better. Thank you for pointing this.

Regarding the default value, I like clean code either. I'd appreciate if you could clarify why empty string is better than None.

Reply:
@zhan9san,
when you use empty string you show to the user that string is expected as input. I assume here it could be only string

if any other format might be expected, then None is better

Reply:
Hi @beliaev-maksim 

This issue exists in many functions, like `email` and `password` in `User.__init__`, `username` and `token_id` in `Token.__init__`.

`None` indicates that no value was passed for the optional parameter.

What if empty string is a valid value?

Reply:
you cannot compare all functions the same way

in this particular example we expect only strings and you do a check if checksum value is empty or not

Reply:
I am trying to follow the same code of conduct as how we handle default value of key parameter in other function.
The above examples indicate None is commonly used.

Actually, if i say, "it does a check if the value is set or not" also makes sense.

Both "empty string" and None works for this specific case.

I'd appreciate if you could provide more detailed explanation why we use empty string here and None in other functions.

Reply:
None in other functions because it was done that since years. If we migrate to future version of Python we even may want to use Typing and to use some linter for data type checks.

so in this particular case we can have None, since the whole lib is using it. Consider doing other changes from the review